### PR TITLE
Limit the number of in-flight tasks for loading outdated parts

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
On big servers with big amount of outdated parts (> 10k) caused by many tables of whatever server could receive an exception from ThreadPool which ends up with forceful termination. This behaviour is now fixed.

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
Extracted a part of changed from https://github.com/ClickHouse/ClickHouse/pull/49474 which is on-hold now...

**Review by pufit — APPROVED** [review]

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit 877066cb1140d4d3ea0425752151ac9c025dfba5 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50450/877066cb1140d4d3ea0425752151ac9c025dfba5/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: programs/server/Server.cpp [hunk 1]
```diff
@@ -696,7 +696,7 @@ try
     OutdatedPartsLoadingThreadPool::initialize(
         server_settings.max_outdated_parts_loading_thread_pool_size,
         0, // We don't need any threads one all the parts will be loaded
-        server_settings.outdated_part_loading_thread_pool_queue_size);
+        server_settings.max_outdated_parts_loading_thread_pool_size);
```
Comment (tavplubix) [inline programs/server/Server.cpp:699]: So, `max_threads` does not actually limit the number of threads in ThreadPool? Why does the queue size affect the number of threads?

Comment (nikitamikhaylov) [reply programs/server/Server.cpp:699]: The problem is that `scheduleOrThrow` method throw an exception when there is no free slot in queue, which is 10k (makes no sense). We need such a behaviour to wait until there will be a free slot, which is guaranteed by using `scheduleOrThrowOnError`.

**Pull request merged**